File: ShopSphere-Frontend/src/services/cartService.js
import axiosInstance from "../api/axiosInstance";

export const addToCart = async (productId, quantity = 1) => {
  const response = await axiosInstance.post("/cart/items", {
    productId,
    quantity,
  });

  return response.data;
};

export const getCart = async () => {
  const response = await axiosInstance.get("/cart");

  return response.data;
};

export const updateCartItem = async (cartItemId, quantity) => {
  const response = await axiosInstance.put(`/cart/items/${cartItemId}`, null, {
    params: {
      quantity,
    },
  });

  return response.data;
};

export const removeCartItem = async (cartItemId) => {
  const response = await axiosInstance.delete(`/cart/items/${cartItemId}`);

  return response.data;
};

File: ShopSphere-Backend/src/main/java/com/shopsphere/service/CartService.java
package com.shopsphere.service;

import com.shopsphere.dto.AddToCartRequest;
import com.shopsphere.dto.CartResponse;

public interface CartService {

    CartResponse addToCart(
            Long userId,
            AddToCartRequest request);

    CartResponse getCart(Long userId);

    CartResponse updateCartItem(
            Long userId,
            Long cartItemId,
            Integer quantity);

    CartResponse removeCartItem(
            Long userId,
            Long cartItemId);
}